Fixed compatibility with Symfony 5

Steveb-p opened the pull request:
This PR is a follow up for #12.

### Also adds Github Action configuration for automatic tests.

It provides the following changes:
* Fixes base issues with tests (I need to revisit them once more to provide more sane defaults for config arrays in actual implementations)
* Adds missing `symfony/yaml` dependency (we could also try moving over to PHP/XML service declarations)
* Fixes Command reliance on Container. Now dependencies are injected via constructor, just like in Controller.
* Adds a virtual file system for one of the tests.
* Adds missing ext-curl dev dependency.
* Fixed DataCollector serialization tests.
* Fixed lowest dependency versions (issues detected when installing `composer update --prefer-lowest`, and testing)

Steveb-p committed:
Added missing dependencies, fixed tests, fixed command

Steveb-p committed:
Add Github workflows

Steveb-p committed:
Added default config values to Command & Controller

Steveb-p committed:
Added missing cURL extension required for tests

Steveb-p committed:
Added virtual file system for tests

Steveb-p committed:
Fixed CurlTest failure

Steveb-p committed:
Fixed DoctrineYumlDataCollectorTest serialization failure

Steveb-p committed:
Added proper tests autoloading declaration to composer

Steveb-p committed:
Added Config class to reduce code duplication

Steveb-p committed:
Fix lowest dependencies

Steveb-p committed:
Added missing newline and cs script

Nono1971 commented:
seems already great !
Are all tests OK for you or do you need more time ? (don't have much time left actually to test by myself)
tell me when (or if) you think it is Ok for you :)
Thanks a lot for your contribution, it helps a lot !

Steveb-p commented:
@Nono1971 I'll look at it again tonight. I had a few thoughts at the end that I wanted to look into.

Are you ok with switching to GH actions, or should I put into a second PR? (They need to be enabled in repository settings too).

Nono1971 commented:
OK, I let you see tonight :)

For GH actions, I'm neither OK or KO, I just didn't think about it yet. If you have propositions about it, why not ! (don't know if it would be worth or not yet... but it might be interesting to look at it).
Anyway, it would effectively be better into another PR :) Maybe could you open an issue for this evolution to talk together about the possibilities before working on it ^-^

Steveb-p commented:
Sorry, didn't manage to pick it up yesterday evening. Will try to today.

Regarding GH actions I'll move it to a separate PR. I'm pretty sure it's something you'd like to have since it uses the Github infrastructure (for open source), can report directly into the PR and since it works on containers it's faster (Travis spins up virtual machines for each test, and they recently introduced some limitations into their free, OS version afaik).

Steveb-p committed:
Remove Github Actions

Steveb-p committed:
Fix PHPUnit 8.0 compatibility

Required for future PHP 8.0 compatibility

Steveb-p committed:
Added more PHP & Symfony versions to testing matrix

Steveb-p committed:
Fixed code style warnings

Steveb-p committed:
Added Kernel test

Steveb-p commented:
@Nono1971 Github actions are removed via 4027aa8.

I've up'ed the PHPUnit version requirement to `^8.0` for future PHP 8.0 compatibility / tests. Symfony 4.x tests will fail against PHP 8 atm.

Added more PHP & Symfony versions to testing matrix.

Code style sniffer version has been up'ed to `3.4`. Previous versions fail on PHP 7.4.

Lastly, I've added a Kernel integration test that ensures that the DI container will properly compile in a smallest possible application. Unfortunately due to issues with a warning during compilation:
```
There was 1 error:

1) OnurbTest\Bundle\YumlBundle\KernelTest::testKernelCompiles
"continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?
```
I mark the test as skipped for Symfony 4.x.

Steveb-p committed:
Add missing newline

Steveb-p committed:
Fix travis >7.2

Steveb-p committed:
Allow older version of virtual file system mock

Steveb-p committed:
Allow older version of virtual file system mock

Steveb-p committed:
Removed PHP 7.1 with Symfony 5.x from test matrix

Symfony 5.x does not support PHP 7.1

Steveb-p commented:
Fixed issues shown in tests. Now they pass properly: https://travis-ci.org/github/Nono1971/doctrine-yuml-bundle/builds/764642437

Ideally PR's with failing tests should not be allowed to be merged I guess, but that goes into repository configuration and travis integration :smile:

andrelec1 commented:
???

ahmed-bhs commented:
Up!!
